File: A_Police_Recruits.cpp
#include<bits/stdc++.h>
using namespace std;
const int mx=1e5+123;
int arr[mx];
int main(){
    int n;
    cin>>n;
    for(int i=0;i<n;i++) cin>>arr[i];
    int cnt=0,sum=0;
    for(int i=0;i<n;i++){
        if(arr[i]==-1){
            if(sum!=0){
                sum+=arr[i];
            }
            else {
                cnt++;
            }
        }else{
            sum+=arr[i];
        }
    }
    cout<<cnt<<endl;
    return 0;
}

File: A_Police_Recruits.java
import java.util.Scanner;

public class A_Police_Recruits {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        int cnt = 0;
        int sum = 0;
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        /* for (int i = 0; i < n; i++) {
            System.out.println(arr[i]+" ");
        } */
        for (int i = 0; i < n; i++) {
            if (arr[i] == -1) {
                if (sum > 0) {
                    sum +=arr[i];
                } else
                    cnt++;
            } else {
                sum +=arr[i];
            }
        }
        System.out.println(cnt);
        sc.close();
    }
}